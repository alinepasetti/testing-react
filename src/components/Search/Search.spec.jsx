import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Search } from '.';
import renderer from 'react-test-renderer';

describe('<Search />', () => {
  it('should call the handler on each key pressed', () => {
    const onChangeHandler = jest.fn();
    render(<Search onChangeHandler={onChangeHandler} searchValue="" />);
    const input = screen.getByRole('searchbox');
    expect(input).toBeInTheDocument();

    userEvent.type(input, 'banana');
    expect(onChangeHandler).toBeCalledTimes(6);
  });

  it('should display the received searchValue', () => {
    const onChangeHandler = jest.fn();
    render(<Search onChangeHandler={onChangeHandler} searchValue="banana" />);
    const input = screen.getByRole('searchbox');
    expect(input.value).toBe('banana');
  });

  it('should match snapshot', () => {
    const onChangeHandler = jest.fn();
    const postsElement = renderer.create(<Search onChangeHandler={onChangeHandler} searchValue="banana" />).toJSON();
    expect(postsElement).toMatchSnapshot();
  });
});
