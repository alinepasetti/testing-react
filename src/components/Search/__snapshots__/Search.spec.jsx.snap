// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should match snapshot 1`] = `
<div
  className="search-container"
>
  <input
    onChange={[MockFunction]}
    placeholder="Type your search"
    type="search"
    value="banana"
  />
</div>
`;
